packages/nimble/inst/tests/test-nimbleList.R

```diff
@@ -1296,13 +1306,68 @@ expect_equal(RnimbleList$v, CnimbleList$v)
 expect_equal(testInst$testList27$testSvd$d, CtestInst$testList27$testSvd$d)
 expect_equal(testInst$testList27$testSvd$u, CtestInst$testList27$testSvd$u)
 expect_equal(testInst$testList27$testSvd$v, CtestInst$testList27$testSvd$v)
+test_that("return object (from c++) is nimbleList.",
```

Comment:
nit: (and beyond scope of this PR) It is safer to wrap the entire test in a `test_that`, e.g.
```r
test_that('nimbleList Test 27', {
  nlTetsFunc <- nimbleFunction(
    ...
  )
  testInst <- ...
  ...
  expect_equal(diag(1), ...)
  ...
  expect_identical(is.nl(CnimbleList), TRUE, info = 'Return object (from c++) is nimbleList')
})
```
This pattern has the following advantages:
- avoids copy-and-paste errors
- avoids polluting global environment
- is compatible with both `Rscript test-nimbleList.R` and `test_package('nimble', 'nimbleList')`. (The current pattern will stop at the first error in the `Rscript ...` version, and will ignore some failed expectations).
- Correctly attributes errors in setup (e.g. if nlTestFunc27 errors, it will be reported correctly).